src/account-mgr.cpp

```diff
@@ -148,17 +148,14 @@ bool AccountManager::setCurrentAccount(const Account& account)
 
 int AccountManager::replaceAccount(const Account& old_account, const Account& new_account)
 {
-    int i = 0;
-    for (i = 0; i < accounts_.size(); i++) {
-        if (accounts_[i].serverUrl.toString() == old_account.serverUrl.toString()
-            && accounts_[i].username == old_account.username) {
-            accounts_.erase(accounts_.begin() + i);
+    for (size_t i = 0; i < accounts_.size(); i++) {
+        if (accounts_[i] == old_account) {
```

Comment:
Should not use `==` here. The `operator==` also compares the token, which we should not do here.